crates/services/gas_price_service/src/fuel_gas_price_updater/fuel_da_source_adapter/block_committer_costs.rs

```diff
@@ -0,0 +1,60 @@
+use crate::fuel_gas_price_updater::{
+    service::Result as DaBlockCostsResult,
+    DaBlockCosts,
+    DaBlockCostsSource,
+};
+use anyhow::anyhow;
+use reqwest::Url;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+/// This struct is used to denote the block committer da gas price source,,
+/// which receives data from the block committer (only http api for now)
+pub struct BlockCommitterDaBlockCosts {
+    client: reqwest::Client,
+    url: Url,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, Default, PartialEq)]
+struct RawDaBlockCosts {
+    pub l2_block_range: core::ops::Range<u32>,
```

Comment:
I think we just want to know what is the last L2 height is and what the L1 height used to commit the last blob.

Reply:
the block committer will return unordered data to us. they will commit a set of L2 blocks at once to the L1, and return the range, the cost and the size of the blob :)

Reply:
It looks like it would be simple if they just returned the last DA block height and the last committed L2 block height. Since they commit block only in order

Reply:
If the committer does not commit in order I don't understand how fraud proving will work and how L1 contract can know which header belongs to which block.

Reply:
We've also solidified some design decisions that I might've not communicated since my last discussion with @MitchTurner and @rymnc .

The workflow for posting L2 data will roughly look like this:
we accumulate some amount of blocks from L2 into a bundle -> we apply compression over this bundle -> the resulting compressed data is then fragmented to fit into blob transactions (+ some metadata) and sent to L1.

The final order of those transactions shouldn't matter for the following reasons
- the data is compressed and the order can be reconstructed from the metadata
- the endpoint for the gas algorithm will only provide data for a full range of posted blocks

Reply:
> the data is compressed and the order can be reconstructed from the metadata

What if the first half of the block at height X in another blob and you haven't submitted it yet? How can we reconstruct block without the first half?

> we apply compression over this bundle

Do you mean GZIP? Because if you mean DA compression that we work on, I think we will just provide a new endpoint for you to fetch compressed blocks=)

Reply:
> What if the first half of the block at height X in another blob and you haven't submitted it yet? How can we reconstruct block without the first half?

The idea is that you always reconstruct in bundles. The last fragment of the bundle, submitted in the last tx for that bundle can contain metadata like the number of fragments, compression metadata, the block height at which we sent the first fragment etc.
For reconstruction you would need to fetch all the fragments belonging to the bundle and decompress

> Do you mean GZIP? Because if you mean DA compression that we work on, I think we will just provide a new endpoint for you to fetch compressed blocks=)

I wasn't aware that there will be an endpoint for compressed blocks. Can we still expect to make gains if we also do something like GZIP on our end?